src/ecs/components/rendering/texture.rs

```diff
@@ -0,0 +1,235 @@
+            TextureError::UnsupportedColorFormat(col_fmt) => {
+                write!(fmt, "Unsupported color format: {:?}", col_fmt)
+            }
+            TextureError::UnsupportedSize { max, got } => {
+                write!(fmt, "Unsupported size. max: {:?}, got: {:?}", max, got)
+            }
+        }
+    }
+}
+
+
+enum Inner {
+    Factory(FactoryFuture<Texture, RendererError>),
+    Err(Option<TextureError>),
+}
+
+/// Will be `TextureComponent` result type of `TextureContext::create_asset`
+pub struct TextureFuture(Inner);
+
+impl TextureFuture {
+    fn factory(future: FactoryFuture<Texture, RendererError>) -> Self {
+        TextureFuture(Inner::Factory(future))
+    }
+
+    fn from_error(error: TextureError) -> Self {
+        TextureFuture(Inner::Err(Some(error)))
+    }
+
+    fn unsupported_color_format(fmt: ColFmt) -> Self {
+        Self::from_error(TextureError::UnsupportedColorFormat(fmt))
+    }
+
+    fn unsupported_size(width: usize, height: usize) -> Self {
+        Self::from_error(TextureError::UnsupportedSize {
+                             max: (u16::max_value() as usize, u16::max_value() as usize),
+                             got: (width, height),
+                         })
+    }
+}
+
+impl Future for TextureFuture {
+    type Item = TextureComponent;
+    type Error = TextureError;
+
+    fn poll(&mut self) -> Poll<TextureComponent, TextureError> {
+        match self.0 {
+            Inner::Factory(ref mut future) => {
+                match future.poll() {
+                    Ok(Async::NotReady) => Ok(Async::NotReady),
+                    Ok(Async::Ready(texture)) => Ok(Async::Ready(TextureComponent::new(texture))),
+                    Err(err) => Err(TextureError::Renderer(err)),
+                }
+            }
+            Inner::Err(ref mut err) => Err(err.take().expect("polling completed future")),
+        }
+    }
+}
+
+
+
+/// Wraps `Texture` into component
+#[derive(Clone, Debug)]
+pub struct TextureComponent(pub AssetPtr<Texture, TextureComponent>);
+impl AsRef<Texture> for TextureComponent {
+    fn as_ref(&self) -> &Texture {
+        self.0.inner()
+    }
+}
+impl AsMut<Texture> for TextureComponent {
+    fn as_mut(&mut self) -> &mut Texture {
+        self.0.inner_mut()
+    }
+}
+
+impl TextureComponent {
+    fn new(texture: Texture) -> Self {
+        TextureComponent(AssetPtr::new(texture))
+    }
+}
+
+impl Component for TextureComponent {
+    type Storage = VecStorage<Self>;
+}
+
+impl Asset for TextureComponent {
+    type Context = TextureContext;
+}
+
+/// Context to create textures from images
+pub struct TextureContext {
+    cache: Cache<AssetFuture<TextureComponent>>,
+    factory: Arc<Factory>,
+}
+
+impl TextureContext {
+    pub(crate) fn new(factory: Arc<Factory>) -> Self {
+        TextureContext {
+            cache: Cache::new(),
+            factory: factory,
+        }
+    }
+}
+
+impl Context for TextureContext {
```

Comment:
I see using `SimpleContext` isn't perfect right now, but maybe we could just move its methods into `impl SimpleContext`, remove the implementation of `Context` and the `create_asset` method so we can just delegate to `SimpleContext` for everything except `create_asset`.